api/test/e2enew/route/route_with_script_luacode_test.go

```diff
@@ -0,0 +1,339 @@
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"script_id\":\"r1\"",
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("hit the route", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world\n",
+		}),
+		table.Entry("update route with valid script_id", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"upstream": {
+						"type": "roundrobin",
+						"nodes": {
+								"` + base.UpstreamIp + `:1981": 1
+						}
+				},
+				"script": "local _M = {} \n function _M.access(api_ctx) \n ngx.log(ngx.WARN,\"hit access phase\") \n end \nreturn _M",
+				"script_id": "r1"
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("update route with invalid script_id - not equal to id", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"upstream": {
+						"type": "roundrobin",
+						"nodes": {
+								"` + base.UpstreamIp + `:1980": 1
+						}
+				},
+				"script": "local _M = {} \n function _M.access(api_ctx) \n ngx.log(ngx.WARN,\"hit access phase\")",
+				"script_id": "not-r1"
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusBadRequest,
+		}),
+		table.Entry("update route with invalid script_id - set script_id but without script", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": {
+							"` + base.UpstreamIp + `:1980": 1
+					}
+				},
+				"script_id": "r1"
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusBadRequest,
+		}),
+		table.Entry("delete the route (r1)", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("hit the route just delete", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+			Sleep:        base.SleepTime,
+		}),
+	)
+	ginkgo.It("verify the log generated by script set in Step-4 above", func() {
+		t := ginkgo.GinkgoT()
```

Comment:
We can reuse the `ginkgo.GinkgoT()`.
So we can put it in front.

Reply:
@nic-chen 
Not all test cases use t, which is usually used table.entry we don't need to use t.
But at present, t is used in many places.

Reply:
I mean you can put it in front.
So line 142 and line 327 can merge into one.

Reply:
done. Thks.